src/libraries/System.Private.CoreLib/src/System/SearchValues/IndexOfAnyAsciiSearcher.cs

```diff
@@ -1017,73 +1100,682 @@ public static int LastIndexOfAny<TNegator>(ref byte searchSpace, int searchSpace
+        [CompExactlyDependsOn(typeof(PackedSimd))]
+        private static TResult IndexOfAnyCore<TResult, TNegator, TResultMapper>(ref short searchSpace, int searchSpaceLength, ref UniqueLowNibbleState state)
+            where TResult : struct
+            where TNegator : struct, INegator
+            where TResultMapper : struct, IResultMapper<short, TResult>
         {
-            // On X86, the Ssse3.Shuffle instruction will already perform an implicit 'AND 0xF' on the indices, so we can skip it.
-            // For values above 127, Ssse3.Shuffle will also set the result to 0. This is fine as we don't want non-ASCII values to match anyway.
-            Vector128<byte> lowNibbles = Ssse3.IsSupported
-                ? source
-                : source & Vector128.Create((byte)0xF);
+            ref short currentSearchSpace = ref searchSpace;
 
-            // On ARM, we have an instruction for an arithmetic right shift of 1-byte signed values.
-            // The shift will map values above 127 to values above 16, which the shuffle will then map to 0.
-            // On X86 and WASM, use a logical right shift instead.
-            Vector128<byte> highNibbles = AdvSimd.IsSupported
-                ? AdvSimd.ShiftRightArithmetic(source.AsSByte(), 4).AsByte()
-                : source >>> 4;
+            if (searchSpaceLength < Vector128<ushort>.Count)
+            {
+                ref short searchSpaceEnd = ref Unsafe.Add(ref searchSpace, searchSpaceLength);
 
-            // The bitmapLookup represents a 8x16 table of bits, indicating whether a character is present in the needle.
-            // Lookup the rows via the lower nibble and the column via the higher nibble.
-            Vector128<byte> bitMask = Vector128.ShuffleUnsafe(bitmapLookup, lowNibbles);
+                while (!Unsafe.AreSame(ref currentSearchSpace, ref searchSpaceEnd))
+                {
+                    char c = (char)currentSearchSpace;
+                    if (TNegator.NegateIfNeeded(state.Lookup.Contains128(c)))
+                    {
+                        return TResultMapper.ScalarResult(ref searchSpace, ref currentSearchSpace);
+                    }
 
-            // For values above 127, the high nibble will be above 7. We construct the positions vector for the shuffle such that those values map to 0.
-            Vector128<byte> bitPositions = Vector128.ShuffleUnsafe(Vector128.Create(0x8040201008040201, 0).AsByte(), highNibbles);
+                    currentSearchSpace = ref Unsafe.Add(ref currentSearchSpace, 1);
+                }
 
-            Vector128<byte> result = bitMask & bitPositions;
-            return result;
+                return TResultMapper.NotFound;
+            }
+
+#pragma warning disable IntrinsicsInSystemPrivateCoreLibAttributeNotSpecificEnough // The behavior of the rest of the function remains the same if Avx2.IsSupported is false
+            if (Avx2.IsSupported && searchSpaceLength > 2 * Vector128<short>.Count)
```

Comment:
No specialization for AVX512?

Reply:
Not yet*. I wanted to avoid any potential regressions that we've hit with previous attempts at Avx512 here.


There are, however, even more interesting things we can do with Avx512.
The approach this PR is adding is using the low nibble (4 bits) because 4-bit controls are what `Shuffle`s are capable of using (`Vector128.ShuffleUnsafe`, `Avx2.Shuffle`).
With `Avx512Vbmi` though, we can use `PermuteVar64x8` instead, allowing us to still do the lookup with 2 instructions, but now for any values with the lower 6 bits being unique.
```c#
Vector512<byte> values = Avx512Vbmi.PermuteVar64x8(valuesByLow6Bits, source);
return Vector512.Equals(source, values);
```

Looking at the "real world regexes", there are 1696 unique `SearchValues<char>` inputs.
Out of those, 1323 are fully ASCII.
284 have unique low 4 bits.
696 have unique low 6 bits, which includes sets like `[a-zA-Z]`.

I still have to do a bunch of measuring for that though, but soon ™️ :)